ADC/Core/Src/init.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * init.c
+ *
+ *  Created on: Mar 10, 2020
+ *      Author: rodrigocelmer
+ */
+#include "stm32l053xx.h"
+
+void ADC_Init (void)
+{
+	/*
+	 * since we're going to initialize and then calibrate the ADC, we have to be sure that:
+	 * 1) ADEN = 0,
+	 * 2) DMAEN = 0,
+	 * 3) ADSTART = 0 and
+	 * 4) ADRDY = 0
+	 */
+	if(ADC1->CR & ADC_CR_ADSTART)			//checking if there's an ongoing conversion (3)
```

Comment:
if the processor is coming from a reset, why do we need to check if there is any conversion ongoing?

Reply:
Apple asked to do so:
"gustavoscherer 14 days ago  Collaborator
I'm not sure, but maybe you should check the ADSTART bit, before set the LFMEN, cause It can be written only when ADSTART is 0."
By the way, all those registers' reset values are zero, I don't think it's quite necessary to do so during the initialization process

Reply:
I asked to do it just before enable LFMEN, cause the RM says this bit can only be written when ADSTART=0. And in my opinion, is good to check if there is any conversion ongoing, cause you can call this function any time in the code and not only when the processor is coming from a reset.

Reply:
@gustavoscherer the initialization function for any peripheral, I would call it only once, at the very start of the firmware, differently from ADC_Config, for example. But I understand your point
@Vinimuller how should I keep the init function?

Reply:
If you call init function once in the code, I agree you don't need the lines before you set the clock for the ADC.

Reply:
For the sake of this example let's remove it. Although @gustavoscherer is right. You may call this function after waking up from sleep, then it would be a good practice to check if any conversion is ongoing.